gapis/gfxapi/synchronization_data.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gfxapi
+
+import "sort"
+
+type SynchronizationIndex uint64
+type SynchronizationIndices []SynchronizationIndex
+type SubcommandIndex []uint64
+
+type ExecutionRanges struct {
+	LastIndex SubcommandIndex
+	Ranges    map[SynchronizationIndex]SubcommandIndex
+}
+
+type SynchronizationData struct {
+	CommandRanges map[SynchronizationIndex]ExecutionRanges
+}
+
+func NewSynchronizationData() *SynchronizationData {
+	s := new(SynchronizationData)
+	s.CommandRanges = make(map[SynchronizationIndex]ExecutionRanges)
+	return s
+}
+
+func (s SubcommandIndex) LessThan(s2 SubcommandIndex) bool {
+	for i := range s {
+		if i > len(s2)-1 {
+			break
+		}
+		if s[i] < s2[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *SubcommandIndex) Decrement() {
```

Comment:
Please can we have some basic tests on this? Sort of thing that can hide bugs which takes days to trace.